salt/states/blockdev.py

```diff
@@ -154,20 +150,29 @@ def formatted(name, fs_type='ext4', **kwargs):
             cmd += '-i size={0} '.format(kwargs['inode_size'])
     cmd += name
     __salt__['cmd.run'](cmd).splitlines()
-    __salt__['cmd.run']('sync').splitlines()
-    blk = __salt__['cmd.run']('lsblk -o fstype {0}'.format(name)).splitlines()
 
-    if len(blk) == 1:
-        current_fs = ''
-    else:
-        current_fs = blk[1]
-
-    if current_fs == fs_type:
-        ret['comment'] = ('{0} has been formatted '
-                          'with {1}').format(name, fs_type)
-        ret['changes'] = {'new': fs_type, 'old': current_fs}
-        ret['result'] = True
-    else:
-        ret['comment'] = 'Failed to format {0}'.format(name)
-        ret['result'] = False
+    # Repeat lsblk check up to 10 times with 3s sleeping between each
+    # to avoid lsblk failing although mkfs has succeeded
+    # see https://github.com/saltstack/salt/issues/25775
+    for i in range(10):
+        current_fs = _checkblk(name)
+
+        if current_fs == fs_type:
+            ret['comment'] = ('{0} has been formatted '
+                              'with {1}').format(name, fs_type)
+            ret['changes'] = {'new': fs_type, 'old': current_fs}
+            ret['result'] = True
+            return ret
```

Comment:
What do you think about logging here? I think it would be useful to know which iteration we're on and give the user some indication of what's happening. A 30 second unexplained sleep would be hard to diagnose otherwise. :]